python/metatensor-learn/tests/equivariant_transform.py

```diff
@@ -0,0 +1,79 @@
+import os
+
+import pytest
+
+import metatensor
+
+
+torch = pytest.importorskip("torch")
+
+from metatensor.learn.nn.equivariant_transform import EquivariantTransform  # noqa: E402
+
+from ._rotation_utils import WignerDReal  # noqa: E402
+
+
+DATA_ROOT = os.path.join(
+    os.path.dirname(__file__), "..", "..", "metatensor-operations", "tests", "data"
+)
+
+
+@pytest.fixture
+def tensor():
+    tensor = metatensor.load(os.path.join(DATA_ROOT, "qm7-spherical-expansion.npz"))
+    tensor = tensor.to(arrays="torch")
+    tensor = metatensor.remove_gradients(tensor)
+    return tensor
+
+
+@pytest.fixture
+def wigner_d_real():
+    return WignerDReal(lmax=4, angles=(0.87641, 1.8729, 0.9187))
+
+
+def module_wrapper(in_features, device, dtype):
+    """
+    A sequential module with nonlinearities
+    """
+    return torch.nn.Sequential(
+        torch.nn.Tanh(),
+        torch.nn.Linear(
+            in_features=in_features,
+            out_features=5,
+            device=device,
+            dtype=dtype,
+        ),
+        torch.nn.ReLU(),
+        torch.nn.Linear(
+            in_features=5,
+            out_features=in_features,
+            device=device,
+            dtype=dtype,
+        ),
+    )
+
+
+def test_equivariance(tensor, wigner_d_real):
```

Comment:
Looks good! Maybe we could also have a test with a module that uses the identity operation (`torch.nn.Identity`) and check input equals output?